packages/mobile/e2e/src/usecases/WalletConnectV1.js

```diff
@@ -0,0 +1,324 @@
+  afterEach(async () => {
+    // If on details page go back
+    let backChevronPresent = await isElementVisible('BackChevron')
+    if (backChevronPresent) {
+      await element(by.id('BackChevron')).tap()
+    }
+    // Cancel pending wc action
+    let closeIconPresent = await isElementVisible('Times')
+    if (closeIconPresent) {
+      await element(by.id('Times')).tap()
+    }
+  })
+
+  afterAll(async () => {
+    await walletConnector.transportClose()
+  })
+
+  it('Then is able to send a transaction (eth_sendTransaction)', async () => {
+    // Save result and await for it later
+    let result = walletConnector.sendTransaction(tx)
+
+    // Verify transaction type text
+    await waitFor(element(by.text('Send a Celo TX')))
+      .toBeVisible()
+      .withTimeout(10 * 1000)
+
+    // View and assert on Data - TODO Move to Component Tests
+    await element(by.text('Show details')).tap()
+    await expect(element(by.id('Dapp-Data'))).toHaveText(`[${JSON.stringify(tx)}]`)
+    await element(by.id('BackChevron')).tap()
+
+    // Accept and verify UI behavior
+    await element(by.text('Allow')).tap()
+    await enterPinUiIfNecessary()
+    await waitFor(element(by.text('Success! Please go back to WalletConnectV1 E2E to continue')))
+      .toBeVisible()
+      .withTimeout(10 * 1000)
+    await waitFor(element(by.id('SendOrRequestBar')))
+      .toBeVisible()
+      .withTimeout(10 * 1000)
+
+    // Wait for transaction and get receipt
+    let txHash = await result
+    let txReceipt = await kit.connection.getTransactionReceipt(txHash)
+
+    // Assert on transaction receipt
+    jestExpect(txReceipt.status).toStrictEqual(true)
+    jestExpect(txReceipt.from).toStrictEqual(fromAddress)
+    jestExpect(txReceipt.to).toStrictEqual(toAddress)
+  })
+
+  // TODO: Enable when Valora implantation defect is fixed - gas can be optional is resolved
+  it.skip('Then is able to sign a transaction', async () => {
+    // Save result and await for it later
+    let result = walletConnector.signTransaction(tx)
+    await waitFor(element(by.text('Sign a Celo TX')))
+      .toBeVisible()
+      .withTimeout(10 * 1000)
+    await element(by.text('Allow')).tap()
+    await enterPinUiIfNecessary()
+    await waitFor(element(by.id('SendOrRequestBar')))
+      .toBeVisible()
+      .withTimeout(10 * 1000)
+
+    // Wait for signature
+    let signature = await result
+
+    // Verify the signature - what is the default message
+    const valid = verifySignature('0x', signature, fromAddress)
+    jestExpect(valid).toStrictEqual(true)
+  })
+
+  it('Then is able to sign a personal message (personal_sign)', async () => {
+    const message = `My email is valora.test@mailinator.com - ${+new Date()}`
+    const msgParams = [
+      // Both Required
+      utf8ToHex(message),
+      fromAddress,
+    ]
+    let result = walletConnector.signPersonalMessage(msgParams)
+    await waitFor(element(by.text('Sign a data payload')))
```

Comment:
was wondering if you think it'd be a good idea to abstract the logic for lines 165-211 so that it can be reused? just noticed that it's repeated in a few tests, might be easier to maintain if it's centralised